src/qibo/validation/gate_set_tomography.py

```diff
@@ -0,0 +1,1273 @@
+        # Measure states. Measurement basis using function "measure_2qb_states(j, circ, noise_model)".
+        for j in range(0, 16):
+            new_circ = measure_2qb_state_new(j, circ)
+            # expectation_val = GST_measure_2qb(new_circ, k, j, nshots)
+            expectation_val = GST_measure(
+                new_circ,
+                k,
+                j,
+                nshots,
+                nqubits,
+                class_qubit_gate,
+                theta,
+                idx_basis_ops,
+                save_data,
+            )
+            matrix_jk_2qb[j, k] = expectation_val
+
+    return matrix_jk_2qb
+
+
+def GST_1qb_basis_operations(
+    nshots=int(1e4), noise_model=None, gjk_1qb=None, save_data=None, backend=None
+):
+    """Runs gate set tomography for the 13 basis operations for a single qubit.
+
+        Args:
+        nshots (int, optional): Number of shots used in Gate Set Tomography.
+        noise_model (:class:`qibo.noise.NoiseModel`, optional): Noise model applied
+            to simulate noisy computation.
+        gjk_1qb (numpy.matrix): Matrix with elements Tr(Q_j rho_k) for one qubit with no gate.
+            If None, call GST_1qb without gate.
+        save_data: Flag to save gate set tomography data. If None, skip.
+        backend (:class:`qibo.backends.abstract.Backend`, optional): Calculation engine.
+
+        The 13 basis operations are based on Physical Review Research 3, 033178 (2021).
+    Returns:
+        numpy.matrix: Matrix with elements jk equivalent to either Tr(Q_j rho_k) or Tr(Q_j O_l rho_k).
+    """
+
+    if backend is None:  # pragma: no cover
+        backend = GlobalBackend()
+
+    if gjk_1qb is None:
+        gjk_1qb = GST_1qb(nshots, noise_model, save_data=None)
+
+    nqubits = 1
+    class_qubit_gate = None
+    theta = None
+
+    BasisOps_13 = [
+        gates.I(0).matrix(backend=GlobalBackend()),
+        gates.X(0).matrix(backend=GlobalBackend()),
+        gates.Y(0).matrix(backend=GlobalBackend()),
+        gates.Z(0).matrix(backend=GlobalBackend()),
+        1 / np.sqrt(2) * np.array([[1, 1j], [1j, 1]]),
+        1 / np.sqrt(2) * np.array([[1, 1], [-1, 1]]),
+        1 / np.sqrt(2) * np.array([[(1 + 1j), 0], [0, (1 - 1j)]]),
+        1 / np.sqrt(2) * np.array([[1, -1j], [1j, -1]]),
+        gates.H(0).matrix(backend=GlobalBackend()),
+        1 / np.sqrt(2) * np.array([[0, (1 - 1j)], [(1 + 1j), 0]]),
+        np.array([[1, 0], [0, 0]]),
+        np.array([[0.5, 0.5], [0.5, 0.5]]),
+        np.array([[0.5, -0.5j], [0.5j, 0.5]]),
+    ]
+
+    ###################################
+    ### 13 scenarios # (2023-11-06) ###
+    ###################################
+
+    gatenames = (
+        "BasisOp00",
+        "BasisOp01",
+        "BasisOp02",
+        "BasisOp03",
+        "BasisOp04",
+        "BasisOp05",
+        "BasisOp06",
+        "BasisOp07",
+        "BasisOp08",
+        "BasisOp09",
+        "BasisOp10",
+        "BasisOp11",
+        "BasisOp12",
+    )
```

Comment:
I guess calling an integer from ```range(13)``` can be used to identify the basis operations. I want to explicitly label the name of each unitary operator as ```BasisOp__``` for clarity. 

When we use ```circuit.raw```, we'll see this:
```
{'name': 'BasisOp05', 'init_args': [array([[ 0.70710678,  0.70710678],
       [-0.70710678,  0.70710678]]), 0], 'init_kwargs': {}, '_target_qubits': (0,), '_control_qubits': [], '_class': 'Unitary'}
```
so that it is clear that we're using ```BasisOp05```. Thanks for bringing up this point. I have spotted a lack of an additional "BasisOp" for the 72 basis operations.